src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/builder/policy/PolicyViolationMessageBuilder.java

```diff
@@ -0,0 +1,201 @@
+            Optional<PolicyRuleView> optionalPolicyRule = blackDuckResponseCache.getPolicyRule(blackDuckResponseCache, policyInfo);
+            if (optionalPolicyRule.isPresent() && optionalBomComponent.isPresent() && policyCommonBuilder.hasVulnerabilityRule(optionalPolicyRule.get())) {
+                List<ComponentItem> vulnerabilityPolicyItems =
+                    createVulnerabilityPolicyItems(blackDuckResponseCache, blackDuckService, componentService, optionalBomComponent.get(), policyNameItem, nullablePolicySeverityItem, componentName, componentVersionName,
+                        notificationId);
+                componentItems.addAll(vulnerabilityPolicyItems);
+            }
+        }
+        return componentItems;
+    }
+
+    private List<ComponentItem> createVulnerabilityPolicyItems(BlackDuckResponseCache blackDuckResponseCache, BlackDuckService blackDuckService, ComponentService componentService, VersionBomComponentView bomComponent,
+        LinkableItem policyNameItem, LinkableItem policySeverity,
+        String componentName, String componentVersionName, Long notificationId) {
+        List<ComponentItem> vulnerabilityPolicyItems = new ArrayList<>();
+        Optional<ProjectVersionWrapper> optionalProjectVersionWrapper = blackDuckResponseCache.getProjectVersionWrapper(bomComponent);
+        if (optionalProjectVersionWrapper.isPresent()) {
+            try {
+                ProjectVersionWrapper projectVersionWrapper = optionalProjectVersionWrapper.get();
+                String projectVersionUrl = projectVersionWrapper.getProjectVersionView().getHref().orElse(null);
+                ComponentData componentData = new ComponentData(componentName, componentVersionName, projectVersionUrl);
+                List<VulnerableComponentView> vulnerableComponentViews = vulnerabilityUtil.getVulnerableComponentViews(blackDuckService, projectVersionWrapper, bomComponent);
+                List<ComponentItem> vulnerabilityComponentItems = policyCommonBuilder
+                                                                      .createVulnerabilityPolicyComponentItems(vulnerableComponentViews, policyNameItem, policySeverity, componentData, notificationId,
+                                                                          blackDuckService, blackDuckResponseCache);
+                vulnerabilityPolicyItems.addAll(vulnerabilityComponentItems);
+                ComponentVersionView componentVersionView = blackDuckResponseCache.getItem(ComponentVersionView.class, bomComponent.getComponentVersion()).orElse(null);
+
+                Optional<ComponentItem> remediationComponentItem = createRemediationComponentItem(blackDuckResponseCache, MessageBuilderConstants.CATEGORY_TYPE_POLICY, componentService, componentVersionView,
+                    componentData, policyNameItem,
+                    policySeverity, true,
+                    notificationId);
```

Comment:
The formatting for this call seems a bit strange. Do we want an item per line? only break when we reach the edge of the page?